engine/verification/assigner/blockconsumer/consumer.go

```diff
@@ -78,7 +81,9 @@ func (c *BlockConsumer) NotifyJobIsDone(jobID module.JobID) {
 // The consumer retrieves the new blocks from its block reader module, hence it does not need to use the parameter
 // of OnFinalizedBlock here.
 func (c *BlockConsumer) OnFinalizedBlock(*model.Block) {
-	c.consumer.Check()
+	c.unit.Launch(func() {
+		c.consumer.Check()
+	})
```

Comment:
Not sure if I could get the comment correctly, but this is my understanding of the logic: 

- We have not yet wired the new engines of the verification node (including `BlockConsumer`) in the `main` file of the verification node, i.e., `cmd/verification/main.go`. Once we ultimately upgrade the pipeline with the new engines and add them to the main file, they either appear as a `Module` ([example here](https://github.com/onflow/flow-go/blob/master/cmd/verification/main.go#L262-L265)) or a `Component` ([example here](https://github.com/onflow/flow-go/blob/master/cmd/verification/main.go#L266-L282)).  

- Now, in particular about the `BlockConsumer` since it implements a `ReadyDoneAware` interface, we will introduce and bootstrap it as a `Component`.

- Once all components are set, the node builder invokes its [`Run` method](https://github.com/onflow/flow-go/blob/master/cmd/verification/main.go#L397), which goes through all `Component`s of the node (e.g., `BlockConsumer` and other engines) and [turns them on one by one by invoking their `Ready` method](https://github.com/onflow/flow-go/blob/3e00d66e5c81a9ef724a187bcbd763feacf7c0f3/cmd/scaffold.go#L718-L722).  The same method also listens to the shutdown signal from OS, and once the node is ordered for graceful shutdown, it goes through all **[`Done` methods of its components](https://github.com/onflow/flow-go/blob/3e00d66e5c81a9ef724a187bcbd763feacf7c0f3/cmd/scaffold.go#L724-L736) and shut them down one by one.**

So in conclusion, once we were the new pipeline in, we indeed make the `Ready` and `Done` methods of their components on their startup and shutdown. Now considering this, I feel it makes it more safe to handle `goroutine` invocations within `unit` so that upon a graceful shutdown of the node, we won't be leaving any orphan thread lingering around.

Reply:
Also, regarding 

> Just as a reminder, I didn't see that this point from my last review was addressed (?) jobqueue.Consumer: there is the map processings, to which we consistently add elements. Is the content of this map ever cleaned, or does it keep growing indefinitely?

I have already issued that up here: https://github.com/dapperlabs/flow-go/issues/5486

Reply:
I've resolved this issue in this PR, waiting for review:
https://github.com/onflow/flow-go/pull/656